# Slides for the "Detection of Differentially Interacting Chromatin Regions From Multiple Hi-C Datasets" workshop

[HiCcompareWorkshop](https://github.com/mdozmorov/HiCcompareWorkshop) introduces methods for the comparative (aka differential) analysis of the three-dimensional (3D) structure of the genome using data generated by high-throughput chromatin conformation capture (Hi-C) technologies. Hi-C data allows for insights into the genome-wide 3D genomic interactions which play an important role in regulating gene expression and other genomic processes. Just as differential expression analyses using RNA-seq data have become a routine part of genomic experiments, we expect the differential analysis of genomic interactions to become a common task. This workflow will help a novice learn to perform differential analysis of two or more Hi-C datasets and interpret the differential genomic interactions' results. 

This workshop is based on [Stansfield et al., “R Tutorial: Detection of Differentially Interacting Chromatin Regions From Multiple Hi-C Datasets.”](https://currentprotocols.onlinelibrary.wiley.com/doi/abs/10.1002/cpbi.76). 

- [Slides with a brief introduction for the workshop](https://mdozmorov.github.io/Talk_HiCcompare/index.html)
- [HiCcompareWorkshop GitHub repo](https://github.com/mdozmorov/HiCcompareWorkshop)
- [HiCcompareWorkshop Docker image](https://hub.docker.com/repository/docker/mdozmorov/hiccompareworkshop)
- [HiCcompareWorkshop pkgdown website](https://mdozmorov.github.io/HiCcompareWorkshop/)

This workshop will be presented at the [Bioconductor Virtual Conference 2020](https://bioc2020.bioconductor.org/), July 30, 2020, 10:00-10:55am
